.github/workflows/dependabot-build.yml

```
@@ -0,0 +1,47 @@
+# This workflow rebuilds dist/index.js on opened pull request by Dependabot and adds it to the PR
+name: dependabot-build
+
+# trigger workflow only on opened PRs or on demand
+on: 
+  pull_request:
+    types: [opened]
+
+  workflow_dispatch: 
+
+# permissions needed for git push
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+    name: Build
+    if: github.actor == 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run linter
+        run: npm run lint-no-fix
+
+      - name: Build index.js
+        run: npm run build
+
+      - name: Run tests
+        run: npm test
+
+      - name: push index.js
+        run: |
+          git config --global user.email "dependabot@example.com"
+          git config --global user.name "dependabot"
+          git checkout -b ${{github.head_ref}}
```

```suggestion
          git checkout -b ${{ github.head_ref }}
```
Idiomatic GitHub Actions YAML syntax typically has spaces around the expression in the curly brackets.